pennylane/optimize/nesterov_momentum.py

```diff
@@ -35,34 +37,39 @@ class NesterovMomentumOptimizer(MomentumOptimizer):
         momentum (float): user-defined hyperparameter :math:`m`
     """
 
-    def compute_grad(self, objective_fn, x, grad_fn=None):
+    def compute_grad(self, objective_fn, args, kwargs, grad_fn=None):
         r"""Compute gradient of the objective_fn at at the shifted point :math:`(x -
         m\times\text{accumulation})` and return it along with the objective function
         forward pass (if available).
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            args (tuple(array)): Tuple of NumPy arrays containing the current values for the
+                objection function
+            kwargs (dict): Keywords for the cost function
             grad_fn (function): Optional gradient function of the objective function with respect to
                 the variables ``x``. If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
```

Comment:
(same comment here re: above about being explicit with the required gradient function output)